velox/common/memory/SharedArbitrator.cpp

```diff
@@ -472,6 +502,32 @@ uint64_t SharedArbitrator::reclaimUsedMemoryFromCandidates(
   return freedBytes;
 }
 
+uint64_t SharedArbitrator::reclaimUsedMemoryFromCandidatesByAborting(
+    std::vector<Candidate>& candidates,
+    uint64_t targetBytes) {
+  sortCandidatesByCurrentUsage(candidates);
+
+  int64_t freedBytes{0};
+  for (const auto& candidate : candidates) {
+    VELOX_CHECK_LT(freedBytes, targetBytes);
+    if (candidate.currentBytes == 0) {
+      break;
+    }
+    try {
+      VELOX_MEM_POOL_ABORTED(fmt::format(
+          "Memory pool aborted due to system memory pressure, current usage {}",
+          succinctBytes(candidate.currentBytes)));
+    } catch (VeloxRuntimeError& ex) {
```

Comment:
Why not use abort() method of SharedArbitrator?

Reply:
Change to use SharedArbitrator::abort()